Graph display of benchmark results

divdavem opened the pull request:
This PR is based on top of #13 
It makes it easy to view benchmark results on a graph.
It works in a very simple way: results are exported to the `results.js` file. There is a `results.html` file that allows to view the average time and each test for each framework on different graphs.
~~Also, benchmarks are run in CI and the resulting js file is uploaded with [upload-artifact](https://github.com/actions/upload-artifact).~~ (removed as mentioned in the comments below)
All tests for all frameworks are run multiple times (cf `executions` variable in the `config.ts` file, currently set to `3`), the average time for each framework for all tests is displayed at the end of each execution and the `results.js` file is updated at that time too.

divdavem committed:
Code splitting

divdavem committed:
Run performance tests in isolation

divdavem committed:
Displays summary by framework at the end or when interrupting tests

divdavem committed:
Save results to js file and display them with chart.js

tomByrer commented:
> benchmarks are run in CI

Cool, but this is a problem, since different CPUs & JS runtimes give different results.
EG my tests using Bun in recent-ish Intel i7 than t-b's MacOS M2 using Node.  In both alien-signals took 1st place, but the middle pack numbers were all shuffled.

Unless you want to make a CI to run on 4-8 different setups?

divdavem commented:
@tomByrer Thank you for your feedback.
You're right when saying that running benchmarks in different environments (hardware, JS engine, OS...) gives different results. Even running benchmarks multiple times in the same environment can give different results.
The graph displayed in the [README](https://github.com/transitive-bullshit/js-reactivity-benchmark?tab=readme-ov-file#results) is one particular environment (`M3 Macbook Pro using Node.js v22.10.0`). Even though it is not the environment everyone uses, it gives an idea of how the different js libraries compare to one another and many people refer to it.
I added the command in CI to automatically have some benchmark results easily. Those results are just displayed in the logs and when downloading the uploaded artifact (cf for example [here](https://github.com/transitive-bullshit/js-reactivity-benchmark/actions/runs/12831333905/artifacts/2447476917)). Having a CI with different setups as you suggest is a good idea. And automatically publishing the results to a more visible place (such as [GitHub Pages](https://pages.github.com/)) linked in the README would be nice as well.
But actually, running benchmarks in CI is not the main focus of this PR. This could be done in other PRs. The main focus is to easily display results visually on different graphs after running benchmarks. That is something some people requested (cf #2). Only the last (small) commit in my PR adds the command in CI, and I do not care much if this last commit is discarded (it also slows down the CI).

divdavem commented:
@tomByrer I have removed the commit which adds tests in CI.
I have added an option in `config.js` to specify the number of times all tests are executed, defaulting to 3.
Graphs now display the min, max and average time for each test for each framework.
